balancer/grpclb/grpclb_test.go

```diff
@@ -656,21 +656,22 @@ func (s) TestDropRequest(t *testing.T) {
 	for i := 0; i < 3; i++ {
 		var p peer.Peer
 		if _, err := testC.EmptyCall(ctx, &testpb.Empty{}, grpc.WaitForReady(true), grpc.Peer(&p)); err != nil {
-			t.Errorf("%v.EmptyCall(_, _) = _, %v, want _, <nil>", testC, err)
+			t.Fatalf("%v.EmptyCall(_, _) = _, %v, want _, <nil>", testC, err)
 		}
-		if want := tss.bePorts[1]; p.Addr.(*net.TCPAddr).Port != want {
-			t.Errorf("got peer: %v, want peer port: %v", p.Addr, want)
+
+		if want := tss.bePorts[1]; p.Addr == nil || p.Addr.(*net.TCPAddr).Port != want {
```

Comment:
We shouldn't need the `p.Addr` check now, which was the reason to switch to `Fatal` instead of `Error`.

Reply:
Ah I see. Yes we can have different error handling.